src/candy_machine/machine.py

```diff
@@ -0,0 +1,106 @@
+"""
+Candy machine  정의
+
+Description:
+    Candy machine 기능을 구현합니다.
+
+Author:
+    Name: KyungHyun Lim
+    Email: lkh1075@gmail.com
+"""
+
+from candy_machine.interface import State
+from candy_machine.states import (
+    HasQuarterState,
+    NoQuarterState,
+    SoldOutState,
+    SoldState,
+    WinnerState,
+)
+
+
+class CandyMachine:
+    """summary
+    Description:
+        Candy machine action 정의
+    """
+
+    def __init__(self, number_candy: int) -> None:
+        self.soldout_state = SoldOutState(self)
+        self.sold_state = SoldState(self)
+        self.hasquarter_state = HasQuarterState(self)
+        self.noquarter_state = NoQuarterState(self)
+        self.winner_state = WinnerState(self)
+
+        self.state = State()
+        self.count = number_candy
+
+        if self.count > 0:
+            self.state = self.noquarter_state
+        else:
+            self.state = self.soldout_state
+
+    def insert_quarter(self) -> None:
+        """동전 삽입에 따른 상태 변화"""
+        self.state.insert_quarter()
+
+    def eject_quarter(self) -> None:
+        """동전 반환에 따른 상태 변화"""
+        self.state.eject_quarter()
+
+    def turn_crank(self) -> None:
+        """다이얼 회전에 따른 상태 변화"""
+        self.state.turn_crank()
+        self.state.dispense()
+
+    def dispense(self) -> None:
+        """Candy 제공에 따른 상태 변화"""
+        self.state.dispense()
+
+    def release_candy(self) -> None:
+        """Candy 제공에 따른 변화 정의"""
+        if self.count > 0:
+            self.count -= 1
+
+    def refill_candy(self, candy: int) -> None:
+        """Candy 리필"""
+        self.count += candy
+
+    def set_state(self, state: State) -> None:
+        """Candy machine 상태 설정
+
+        Args:
+            state: Candy machine에 셋팅할 상태
+
```

Comment:
```suggestion
        Args:
            state (State): Candy machine에 셋팅할 상태

```
와 같이 힌팅에 대한 정보도 주면 어떨까요?

Reply:
추가해서 수정했습니다.